src/components/Payroll/statusConfig.ts

```diff
@@ -0,0 +1,293 @@
+
+      const deadline =
+        payroll.payrollDeadline instanceof Date
+          ? payroll.payrollDeadline
+          : new Date(payroll.payrollDeadline)
+
+      if (isNaN(deadline.getTime())) return false
+
+      const now = new Date()
+      const timeDiffMs = deadline.getTime() - now.getTime()
+
+      return timeDiffMs < 0
+    },
+  },
+  {
+    name: 'readyToSubmitFallback',
+    badge: {
+      variant: 'info',
+      translationKey: 'readyToSubmit',
+    },
+    condition: payroll => !!payroll.calculatedAt && !payroll.processed,
+  },
+  {
+    name: 'waitingForWireIn',
+    badge: {
+      variant: 'warning',
+      translationKey: 'waitingForWireIn',
+    },
+    condition: (payroll, wireInRequest) =>
+      !!payroll.processed && wireInRequest?.status === WIRE_IN_STATUSES.AWAITING_FUNDS,
+  },
+  {
+    name: 'pendingApproval',
+    badge: {
+      variant: 'warning',
+      translationKey: 'pendingApproval',
+    },
+    condition: (payroll, wireInRequest) =>
+      !!payroll.processed && wireInRequest?.status === WIRE_IN_STATUSES.PENDING_REVIEW,
+  },
+  {
+    name: 'waitingForReverseWire',
+    badge: {
+      variant: 'warning',
+      translationKey: 'waitingForReverseWire',
+    },
+    condition: (payroll, wireInRequest) =>
+      !!payroll.processed && wireInRequest?.status === WIRE_IN_STATUSES.APPROVED,
+  },
+  {
+    name: 'dueInHours',
+    badge: payroll => {
+      const deadline =
+        payroll.payrollDeadline instanceof Date
+          ? payroll.payrollDeadline
+          : new Date(payroll.payrollDeadline!)
+
+      const now = new Date()
+      const timeDiffMs = deadline.getTime() - now.getTime()
+      const msPerHour = 1000 * 60 * 60
+      const hoursDiff = timeDiffMs / msPerHour
+      const hours = Math.ceil(hoursDiff)
+
+      return {
+        variant: 'warning',
+        translationKey: 'dueInHours',
+        translationParams: { hours },
+      }
+    },
+    condition: payroll => {
+      if (!payroll.payrollDeadline || payroll.processed) return false
+
+      const activeProcessingStatuses = [
+        PROCESSING_REQUEST_STATUSES.CALCULATING,
+        PROCESSING_REQUEST_STATUSES.SUBMITTING,
+        PROCESSING_REQUEST_STATUSES.PROCESSING_FAILED,
+      ]
+      if (
+        payroll.processingRequest?.status &&
+        activeProcessingStatuses.includes(
+          payroll.processingRequest.status as (typeof activeProcessingStatuses)[number],
+        )
+      ) {
+        return false
+      }
+
+      const deadline =
+        payroll.payrollDeadline instanceof Date
+          ? payroll.payrollDeadline
+          : new Date(payroll.payrollDeadline)
+
+      if (isNaN(deadline.getTime())) return false
+
+      const now = new Date()
+      const timeDiffMs = deadline.getTime() - now.getTime()
+      const msPerHour = 1000 * 60 * 60
+      const hoursDiff = timeDiffMs / msPerHour
+
+      return timeDiffMs > 0 && hoursDiff < 24
```

Comment:
Same question as above if we can manage this with existing date utilities and if we have properly accounted for time zones